pkg/cmd/util/tokencmd/basicauth.go

```diff
@@ -42,8 +42,8 @@ func (c *BasicChallengeHandler) HandleChallenge(headers http.Header) (http.Heade
 		glog.V(2).Info("already prompted for challenge, won't prompt again")
 		return nil, false, nil
 	}
-	if c.Reader == nil && c.handled {
-		glog.V(2).Info("already handled basic challenge, no reader to alter inputs")
+	if c.handled {
```

Comment:
We are supposed to handle the challenge for a given handler only once no matter what. We will catch nil `Reader` a few lines below, and this also makes the handler exit (preventing the infinite loop) in case we don't miss `username` *and* `password`.